qiskit_experiments/library/characterization/analysis/multi_state_discrimination_analysis.py

```diff
@@ -0,0 +1,147 @@
+from qiskit_experiments.framework.matplotlib import get_non_gui_ax
+from qiskit_experiments.data_processing import SkCLF
+
+from sklearn.inspection import DecisionBoundaryDisplay
+from sklearn.linear_model import SGDClassifier
+
+
+class MultiStateDiscriminationAnalysis(BaseAnalysis):
+    """This class will fit the discriminator to the data."""
+
+    @classmethod
+    def _default_options(cls) -> Options:
+        """Return default analysis options.
+
+        Analysis Options:
+            plot (bool): Set ``True`` to create figure for fit result.
+            ax(AxesSubplot): Optional. A matplotlib axis object to draw.
+            discriminator: The discriminator to classify the data.
+        """
+        options = super()._default_options()
+        options.plot = True
+        options.ax = None
+        options.discriminator = SkCLF(SGDClassifier(loss="modified_huber", max_iter=1000,
+                                                    tol=1e-3))
+        return options
+
+    def _run_analysis(
+            self,
+            experiment_data: ExperimentData,
+    ) -> Tuple[List[AnalysisResultData], List["matplotlib.figure.Figure"]]:
+        """Train a discriminator based on the experiment data.
+        Args:
+            experiment_data: the data obtained from the experiment
+        Returns:
+            The configuration of the trained discriminator and the IQ plot.
+        """
+
+        # number of states and shots
+        n_states = len(experiment_data.data())
+        NUM_SHOTS = len(experiment_data.data()[0]['memory'])
+
+        # Process the data and get labels
+        result_data, fit_state = [], []
+        for i in range(n_states):
+            state_data = []
+            for j in range(NUM_SHOTS):
+                re, im = experiment_data.data()[i]['memory'][j][0]
+                state_data.append(re + 1j * im)
+                fit_state.append(experiment_data.data()[i]['metadata']['label'])
+            result_data.append(np.array(state_data))
+        data = np.concatenate([self._reshape_complex_vec(result) for result in result_data])
+
+        # Train a dsicriminator on the processed data
+        discriminator = self.options.discriminator
+        discriminator.fit(data, fit_state)
+
+        # Create figure
+        if self.options.plot:
+            ax = self.options.get("ax", None)
+            figures = [self._levels_plot(discriminator.discriminator, data,
+                                         n_states, NUM_SHOTS, ax).get_figure()]
+        else:
+            figures = []
+
+        return discriminator.config(), figures  # Make a figure this still needs to be done.
+
+    def _reshape_complex_vec(self, vec) -> List[List[float]]:
+        """Take in complex vector vec and return 2d array w/ real, imag entries. This is needed for the learning.
+        Args:
+            vec (List): complex vector of data
+        Returns:
+            List: vector w/ entries given by (real(vec], imag(vec))
```

Comment:
```suggestion
            List: vector with entries given by [real[vec], imag[vec]].
```

Reply:
Addressed in commit [9390381](https://github.com/Qiskit/qiskit-experiments/pull/936/commits/9390381ce44f62e19b8243b76bd212b37df0fa8f).